framework/mutex/MysqlMutex.php

```diff
@@ -56,12 +56,25 @@ public function init()
      */
     protected function acquireLock($name, $timeout = 0)
     {
-        return $this->db->useMaster(function ($db) use ($name, $timeout) {
-            /** @var \yii\db\Connection $db */
-            return (bool) $db->createCommand(
-                'SELECT GET_LOCK(:name, :timeout)',
-                [':name' => $this->hashLockName($name), ':timeout' => $timeout]
-            )->queryScalar();
+        $hashedLockedName = $this->hashLockName($name);
+        return $this->retryAcquire($timeout, function () use ($hashedLockedName, $timeout) {
+            $isFree = $this->db->useMaster(function ($db) use ($hashedLockedName) {
+                /** @var \yii\db\Connection $db */
+                return (bool) $db->createCommand(
+                    'SELECT IS_FREE_LOCK(:name)',
+                    [':name' => $hashedLockedName]
+                )->queryScalar();
+            });
+            if (!$isFree) {
+                return false;
+            }
+            return $this->db->useMaster(function ($db) use ($hashedLockedName, $timeout) {
```

Comment:
could this be done in one query? 

`SELECT IS_FREE_LOCK(:name) AND GET_LOCK(:name)`

Reply:
Sounds like a solution. @raphaelts3 would you please apply it to the pull request and try it?

Reply:
@cebe It will ignore `$timeout` - right now if you call `acquire($name, 10)` it will wait 10 seconds for lock release, and only if it fails method returns `false`.  With your query it will return `false` immediately.

Reply:
Hm... yeah :(